File: guides/micronaut-microservices-services-discover-eureka/bookrecommendation/kotlin/src/main/kotlin/example/micronaut/BookInventoryClient.kt
/*
 * Copyright 2017-2024 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
//tag::packageandimports[]
package example.micronaut

import io.micronaut.http.MediaType.TEXT_PLAIN
import io.micronaut.http.annotation.Consumes
import io.micronaut.http.annotation.Get
import io.micronaut.http.client.annotation.Client
import io.micronaut.retry.annotation.Recoverable
import reactor.core.publisher.Mono
import jakarta.validation.constraints.NotBlank
//end::packageandimports[]

/*
//tag::harcoded[]
@Client("http://localhost:8082") // <1>
@Recoverable(api = BookInventoryOperations::class)
//end::harcoded[]
*/
//tag::eureka[]
@Client(id = "bookinventory") // <1>
@Recoverable(api = BookInventoryOperations::class)
//end::eureka[]
//tag::clazz[]
interface BookInventoryClient : BookInventoryOperations {

    @Consumes(TEXT_PLAIN)
    @Get("/books/stock/{isbn}")
    override fun stock(@NotBlank isbn: String): Mono<Boolean>
}
//end::clazz[]

File: guides/micronaut-microservices-distributed-tracing-zipkin-opentelemetry/bookrecommendation/python/src/example/micronaut/book_inventory_client.py
from abc import abstractmethod
from typing import Annotated

from jakarta.validation.constraints import NotBlank
from micronaut.core.async_.annotation import SingleResult
from micronaut.http import MediaType
from micronaut.http.annotation import Consumes, Get
from micronaut.http.client.annotation import Client
from micronaut.retry.annotation import Recoverable
from org.reactivestreams import Publisher

from .book_inventory_operations import BookInventoryOperations


@Client("http://localhost:8082")
@Recoverable(api=BookInventoryOperations)
class BookInventoryClient(BookInventoryOperations):

    @Consumes(MediaType.TEXT_PLAIN)
    @Get("/books/stock/{isbn}")
    @SingleResult
    @abstractmethod
    def stock(self, isbn: Annotated[str, NotBlank]) -> Publisher[bool]:
        ...
